src/applications/proxy-rewrite/redirects/index.js

```diff
@@ -58,6 +58,24 @@ const redirectIfNecessary = currentWindow => {
       catchAllRedirectMatch.dest
     }`;
   }
+
+  // Check if redirect destination is to a subdomain
+  const toSubdomainRedirects = crossDomainRedirects?.filter(
+    redirect => redirect?.isToSubdomain,
+  );
+
+  const toSubdomainRedirectMatch = toSubdomainRedirects?.find(redirect => {
+    const isHostMatch = deriveIsHostMatch(redirect, currentWindow);
+    const currentPathMatchesSubdomainSrc =
+      currentWindow.location.pathname === redirect.src.toLowerCase();
```

Comment:
I think this is going to be problematic. We likely want to call `toLowerCase` on both values being compared.